src/Sarif/Core/SarifLog.cs

```diff
@@ -237,5 +248,49 @@ internal static void ComputePolicies(Dictionary<string, FailureLevel> localCache
                 }
             }
         }
+
+        internal static bool ShouldSendLog(SarifLog sarifLog)
+        {
+            if (sarifLog?.Runs?.Count > 0)
+            {
+                foreach (Run run in sarifLog.Runs)
+                {
+                    if (run.Results?.Count > 0)
+                    {
+                        return true;
+                    }
+
+                    if (run.Invocations?.Count > 0)
+                    {
+                        foreach (Invocation invocation in run.Invocations)
+                        {
+                            if (invocation.ToolConfigurationNotifications?.Count > 0)
```

Comment:
>ToolConfigurationNotifications [](http://example.com/codeflow?start=43&length=30)

remove this block. a tool configuration notification is not something we need to receive: this condition indicates that the person running the tool has a problem.

make sense?

1) we receive results because those are findings that will be routed to the code owners
2) we receive tool execution notifications because those will be routed to the tool owner to fix.

we do not receive tool configuration notifications because those are to be resolved by the automation/person running the tool. presumably this can be fixed on their side without a tool change.

Reply:
Got it. Wasn't sure if we want the log for troubleshooting config issues. Removed.